src/lib/support/tests/TestValueParser.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <cmath>
+#include <functional>
+#include <string>
+
+#include <lib/support/UnitTestRegistration.h>
+#include <lib/support/value_parser/ValueParserComplex.h>
+#include <lib/support/value_parser/ValueParserIntegral.h>
+
+#include <nlunit-test.h>
+
+using chip::ByteSpan;
+using chip::CharSpan;
+using chip::Optional;
+using chip::ParseValue;
+using chip::app::DataModel::List;
+using chip::app::DataModel::Nullable;
+
+static constexpr size_t kMaxStringLen = 500;
+
+enum class TestEnum
+{
+    TEST_VALUE_0 = 0,
+    TEST_VALUE_1,
+};
+
+enum class TestBitFlag
+{
+    TEST_BIT0 = 0,
+    TEST_BIT1 = 1,
+    TEST_BIT2 = 2,
+    TEST_BIT3 = 4,
+};
+
+template <typename T>
+bool operator==(const Nullable<T> & lhs, const Nullable<T> & rhs)
```

Comment:
Instead of doing this, we should add an operator== on Nullable that calls up to Optional::operator==.  It not existing is just an oversight.